test-harness/src/matcher/span_matcher.rs

```diff
@@ -0,0 +1,64 @@
+use std::fmt::Display;
+
+use crate::tracing_json::Span;
+
+use super::FieldMatcher;
+
+/// A [`Span`] matcher.
+#[derive(Clone)]
+pub struct SpanMatcher {
+    name: String,
```

Comment:
I noticed that for field values you do a regex match but for span names you do an exact match.  I was wondering if it would be possible to also support a regex match for the span name.

Reply:
Yeah, I just haven't needed it yet. I also don't want to have to remember to write `^run_hooks$` instead of `run_hooks` every time I use it.